kyo-prelude/shared/src/main/scala/kyo/Abort.scala

```diff
@@ -200,6 +200,111 @@ object Abort:
       */
     inline def run[E]: RunOps[E] = RunOps(())
 
+    final class RecoverOps[E](dummy: Unit) extends AnyVal:
+
+        /** Recovers from an Abort failure by applying the provided function.
+          *
+          * This method allows you to handle failures in an Abort effect and potentially continue the computation with a new value. It only
+          * handles failures of type E and leaves panics unhandled (Abort[Nothing]).
+          *
+          * @param onFail
+          *   A function that takes the failure value of type E and returns a new computation
+          * @param v
+          *   The original computation that may fail
+          * @return
+          *   A computation that either succeeds with the original value or the recovered value
+          */
+        def apply[A: Flat, B: Flat, S, ER](onFail: E => B < S)(v: => A < (Abort[E | ER] & S))(
+            using
+            frame: Frame,
+            ct: SafeClassTag[E],
+            reduce: Reducible[Abort[ER]]
+        ): (A | B) < (S & Abort[ER]) =
+            ArrowEffect.handle.catching[
```

Comment:
I couldn't avoid the code repetition because of inference issues that require `asInstanceOf`.